css/css-values/getComputedStyle-calc-mixed-units-003.html

```diff
@@ -0,0 +1,115 @@
+<!DOCTYPE html>
+
+  <meta charset="UTF-8">
+
+  <title>CSS Values Test: computed value of calc() values using multiplication/division with mixed units</title>
+
+  <link rel="help" href="https://www.w3.org/TR/css-values-4/#calc-computed-value">
+
+  <meta name="flags" content="">
+  <meta content="This meta-test checks for the resolution and absolutization of computed values with mixed units to 'px' when multiplication/division is used." name="assert">
+
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+
+  <style>
+  html {
+      font-size: 30px;
+  }
+
+  body {
+      font-size: 16px;
+      line-height: 1.25; /* computed value: 20px */
+      height: 500px;
+      margin: 20px;
+      width: 520px;
+  }
+
+  div#target {
+      height: 100px;
+      width: 100px;
+  }
+  </style>
+
+  <div id="target"></div>
+
+  <script>
+  function startTesting() {
+    var targetElement = document.getElementById("target");
+
+    function verifyComputedStyle(property_name, specified_value, expected_value, description) {
+      test(function() {
+        targetElement.style.setProperty(property_name, "initial");
+        targetElement.style.setProperty(property_name, specified_value);
+        assert_equals(getComputedStyle(targetElement)[property_name], expected_value);
+      }, description);
+    }
+
+    verifyComputedStyle("width", "calc(10% * 8.4 + 2em)", "468px", "testing width: calc(10% * 8.4 + 2em)");
+    /*
+      10% of 520px = 52px
+      52px * 8.4 = 436.8 (but we still add 2em in the formula)
+      2em = 32px
+      Total = 468.8px (rounded to 468px)
```

Comment:
Why the rounding?

```js

<div style="width: 468.8px" id="target"></div>
<script>document.body.append(getComputedStyle(target).width);</script>
```

I get `468.8px` in Firefox, `468.797px` in Blink, and `468.796875px` in Webkit. No rounding.

So can you change the factors to avoid decimals? Since browsers have different subpixel precisions.

Reply:
Resolved in [c4ab98c](https://github.com/web-platform-tests/wpt/pull/42517/commits/c4ab98c20d660961dbfb50e4b341320409f5a541) (with several other fixes)